omniscidb/ResultSetRegistry/ColumnarResults.cpp

```diff
@@ -458,10 +458,11 @@ inline void ColumnarResults::writeBackCell(const TargetValue& col_val,
       for (auto& elem_val : **arr_col_val) {
         write_scalar(&elem_val, offset++, elem_type);
       }
+    } else if (type->isFixedLenArray()) {
       // Put NULL sentinel value for fixed length array
-      if (type->isFixedLenArray() && (*arr_col_val)->size() == 0) {
```

Comment:
Are `(*arr_col_val)->size()` and `(bool)(*arr_col_val)` the same?

Reply:
`*arr_col_val` is `boost::optional`, so `bool` operator checks it is not empty and `operator->` accesses stored value.